smac/epm/base_epm.py

```diff
@@ -12,64 +15,188 @@
 class AbstractEPM(object):
     '''Abstract implementation of the EPM API '''
 
-    def __init__(self, rng):
+    def __init__(self, 
+                 instance_features:np.ndarray=None,
+                 pca_dims:float=None):
         '''
         initialize random number generator
 
         Parameters
         ----------
-        rng : np.random.RandomState
+        instance_features: np.ndarray (I, K)
+            Contains the K dimensional instance features
+            of the I different instances
+        pca_dims: float
+            if set to a float, use PCA to reduce dimensionality of instance features
+            also requires to set n_feats (> pca_dims)
```

Comment:
furthermore, since I would like to avoid to accept floats here since we want to check whether I can apply PCA (#feats > #pca_components). if we accept floats as pca_components, we would need to add an ugly type check